packages/riipen-ui/src/components/Popover.jsx

```diff
@@ -0,0 +1,300 @@
+  getOffsetTop,
+  debounce
+} from "../utils";
+
+class Popover extends React.Component {
+  static propTypes = {
+    /**
+     * The content of the component
+     */
+    children: PropTypes.object,
+
+    /**
+     * Array or string of additional CSS classes to use.
+     *
+     * @type {string | Array}
+     */
+    classes: PropTypes.oneOfType([PropTypes.array, PropTypes.string]),
+
+    /**
+     * Function call to handle clickaway/ close events,
+     * if not provided the anchor element must be removed to clear the popover or use isOpen
+     */
+    onClose: PropTypes.func,
+
+    /**
+     * Either a reference to an anchor element or a function to get the reference
+     */
+    anchorEl: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
+
+    /**
+     * Whether the popover should be displayed
+     */
+    isOpen: PropTypes.bool,
+
+    /**
+     * The marigins of the page the popover should respect
+     */
+    marginThreshold: PropTypes.number,
+
+    /**
+     * The location to attach the content too on the anchor element
+     */
+    anchorPosition: PropTypes.object,
+
+    /**
+     * The location to attach the anchor to on the content element
+     */
+    contentPosition: PropTypes.object,
+
+    /**
+     * Whether to lock the scrollbar when the popover is open
+     */
+    lockScroll: PropTypes.bool,
+
+    /**
+     * The type of element to use at the root
+     */
+    type: PropTypes.string
+  };
+
+  static defaultProps = {
+    lockScroll: true,
+    isOpen: true
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      scrollContainer: null,
+      scrollContainerStyle: null,
+      contentRef: null,
+      positionChangeEventHandler: debounce(this.updatePosition, 10),
+      closeEventHandler: this.closeEvent
+    };
+  }
+
+  componentDidMount() {
+    const { positionChangeEventHandler, closeEventHandler } = this.state;
+    const { anchorEl, lockScroll } = this.props;
+    window.addEventListener("resize", positionChangeEventHandler);
+    window.addEventListener("scroll", positionChangeEventHandler);
+    window.addEventListener("mousedown", closeEventHandler);
+    window.addEventListener("keydown", closeEventHandler);
+
+    if (anchorEl && lockScroll) {
+      this.lockParentScroll();
+    }
+  }
+
+  componentWillUnmount() {
+    const { positionChangeEventHandler, closeEventHandler } = this.state;
+    const { anchorEl, lockScroll } = this.props;
+    window.removeEventListener("resize", positionChangeEventHandler);
+    window.removeEventListener("scroll", positionChangeEventHandler);
+    window.removeEventListener("mousedown", closeEventHandler);
+    window.removeEventListener("keydown", closeEventHandler);
+
+    if (anchorEl && lockScroll) {
+      this.resetParentScroll();
+    }
+  }
+
+  getAnchorEl = () => {
+    const { anchorEl } = this.props;
+    return typeof anchorEl === "function" ? anchorEl() : anchorEl;
+  };
+
+  getPositioningStyle = () => {
```

Comment:
I hope you "borrowed" all of this from MUI :).

Reply:
"borrowed" the idea of how it should work and where to get the values, but this one is overall simpler and easier to follow (in my opinion)